### PR TITLE
Use campaign fixture for automated tests.

### PR DESCRIPTION
**Pull request by DFurnes**
Tests should now be fully self-contained (so they'll run happily on QA, staging, or even a fresh local dev). Next step is to get these guys automatically running on QA again.

Closes JIRA issue [FT-19](https://jira.dosomething.org/browse/FT-19).
For review: @aaronschachter 
## Changes
- Uses campaign text fixture created from `campaign.json` for campaign tests.
- Creates and deletes campaign text fixtures before/after tests run.
- Some minor improvements to tests.
## Screenshot

screen shot 2014-08-19 at 10 44 54 am

### CODE REVIEW

File: bin/ds [hunk 1]
```diff
@@ -407,12 +404,15 @@ function run_tests {
     URL='http://dev.dosomething.org:8888'
   fi
 
+  # Run "before tests" tasks:
+  source $BASE_PATH/scripts/tests/before_tests.sh
+
   # Run tests.
   printf "\n\033[1;33mRunning \"tests/$TEST\"...\033[00m\n"
-  casperjs test $BASE_PATH/tests/$TEST --url=$URL --includes=$BASE_PATH/scripts/tests/common.js
+  casperjs test $BASE_PATH/tests/$TEST --url=$URL --campaign_nid=$CAMPAIGN_NID --includes=$BASE_PATH/scripts/tests/common.js
```
Comment (aaronschachter) [inline bin/ds:412]: where does `$CAMPAIGN_NID` get set?

File: scripts/tests/before_tests.sh [hunk 2]
```diff
@@ -50,6 +57,13 @@ drush_create_test_user QA_TEST_ACCOUNT
 drush_create_test_user QA_TEST_CAMPAIGN_SIGNUP_EXISTING
 drush_create_test_user QA_TEST_CAMPAIGN_ACTION
 
+echo "Creating test campaign node from 'campaign.json' fixture..."
+export CAMPAIGN_NID=$(drush_create_campaign campaign.json)
```
Comment (aaronschachter) [inline scripts/tests/before_tests.sh:61]: oh.. from here?

Comment (DFurnes) [inline scripts/tests/before_tests.sh:61]: Aye! I'm using `source` to run the before & after scripts so that they share the same variable scope.

**Comment by aaronschachter** [comment]
looks nice! :+1:

**Pull request merged**